aspnetcore/fundamentals/hosting.md

```diff
@@ -250,33 +559,360 @@ using (host)
+
+Use an instance of `IRouteBuilder` ([Microsoft.AspNetCore.Routing](https://www.nuget.org/packages/Microsoft.AspNetCore.Routing/)) to use routing middleware:
+
+```csharp
+using (var host = WebHost.Start(router => router
+    .MapGet("hello/{name}", (req, res, data) => 
+        res.WriteAsync($"Hello, {data.Values["name"]}!"))
+    .MapGet("buenosdias/{name}", (req, res, data) => 
+        res.WriteAsync($"Buenos dias, {data.Values["name"]}!"))
+    .MapGet("throw/{message?}", (req, res, data) => 
+        throw new Exception((string)data.Values["message"] ?? "Uh oh!"))
+    .MapGet("{greeting}/{name}", (req, res, data) => 
+        res.WriteAsync($"{data.Values["greeting"]}, {data.Values["name"]}!"))
+    .MapGet("", (req, res, data) => res.WriteAsync("Hello, World!"))))
+{
+    host.WaitForShutdown();
+
+    Console.WriteLine("Press any key to exit...");
+    Console.ReadKey();
+}
+```
+
+Use the following browser requests with the example:
+
+| Request                                    | Response                                 |
+| ------------------------------------------ | ---------------------------------------- |
+| `http://localhost:5000/hello/Martin`       | Hello, Martin!                           |
+| `http://localhost:5000/buenosdias/Catrina` | Buenos dias, Catrina!                    |
+| `http://localhost:5000/throw/ooops!`       | Throws an exception with string "ooops!" |
+| `http://localhost:5000/throw`              | Throws an exception with string "Uh oh!" |
+| `http://localhost:5000/Sante/Kevin`        | Sante, Kevin!                            |
+| `http://localhost:5000`                    | Hello World!                             |
+
+Issuing a break (Ctrl-C/SIGINT or SIGTERM) in the console window bypasses the `WaitForShutdown`. The app displays the `Console.WriteLine` message and waits for a keypress to exit.
+
+**Start(string url, Action<IRouteBuilder> routeBuilder)**
+
+Use a URL and an instance of `IRouteBuilder`:
+
+```csharp
+using (var host = WebHost.Start("http://localhost:8080", router => router
+    .MapGet("hello/{name}", (req, res, data) => 
+        res.WriteAsync($"Hello, {data.Values["name"]}!"))
+    .MapGet("buenosdias/{name}", (req, res, data) => 
+        res.WriteAsync($"Buenos dias, {data.Values["name"]}!"))
+    .MapGet("throw/{message?}", (req, res, data) => 
+        throw new Exception((string)data.Values["message"] ?? "Uh oh!"))
+    .MapGet("{greeting}/{name}", (req, res, data) => 
+        res.WriteAsync($"{data.Values["greeting"]}, {data.Values["name"]}!"))
+    .MapGet("", (req, res, data) => res.WriteAsync("Hello, World!"))))
+{
+    host.WaitForShutdown();
+
+    Console.WriteLine("Press any key to exit...");
+    Console.ReadKey();
+}
+```
+
+Produces the same result as **Start(Action<IRouteBuilder> routeBuilder)**, except the app responds at `http://localhost:8080`.
+
+**StartWith(Action<IApplicationBuilder> app)**
+
+Provide a middleware delegate with `Use`:
```

Comment:
Not quite. What's provided here is an `IApplicationBuilder` with which you can set up your middleware pipeline. You may choose to add a middleware delegate with `Use` but that's not the only way to go about it.